src/hal/src/format.rs

```diff
@@ -148,10 +148,12 @@ bitflags!(
     pub struct ImageFeature: u32 {
         /// Image view can be sampled.
         const SAMPLED = 0x1;
-        /// Image view can be used as storage image.
+        /// Image view can be used as storage image with exclusive read & write access.
         const STORAGE = 0x2;
-        /// Image view can be used as storage image (with atomics).
+        /// Image view can be used as storage image with atomics.
         const STORAGE_ATOMIC = 0x4;
+        /// Image view can be used as storage image with simultaneous read/write access.
+        const STORAGE_READ_WRITE = 0x8;
```

Comment:
We need to ensure that Vulkan flags don't intersect our flags, since as you see the Vulkan backend just casts the value raw... So we should either figure out a scheme where gfx-specific values have a weird mask of 0x1000XXX (or something like that), or totally detach our `ImageFeatures` from Vulkan's, and do the more elaborate translation in the Vulkan backend.

Reply:
Went for detaching it in 8db3137
Took the freedom to express relationships between the flags in code, makes things a little bit more sane imho :)